locator/topology.hh

```diff
@@ -78,6 +80,8 @@ public:
     node(const node&) = delete;
     node(node&&) = delete;
 
+    bool operator==(const node&) const = default;
```

Comment:
Hmm, one of the reasons of using const node* in the first place was to discourage comparing the node object, but rather only the pointer value (which essentially qualified by the topology instance).

Node equivalence can be tested by comparing the host_id, but for the purpose of the unordered set if node regerence wrappers we may want to implement operator== by comparing the objects addresses rather than their members.